### PR TITLE
core: add inline description of TEE session field user_ctx

### PR DESCRIPTION
**Pull request by etienne-lms**
Replace inline comment ??? with an appropriate description of
the user_ctx field which points the the caller user TA context
in case a pseudo TA is invoked from the user TA client.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>

### CODE REVIEW

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: core/include/kernel/tee_ta_manager.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ struct tee_ta_session {
 	bool cancel;		/* True if TAF is cancelled */
 	bool cancel_mask;	/* True if cancel is masked */
 	TEE_Time cancel_time;	/* Time when to cancel the TAF */
-	void *user_ctx;		/* ??? */
+	void *user_ctx;		/* Client TA context or NULL (pTA invoked) */
```
Comment (jenswi-linaro) [inline core/include/kernel/tee_ta_manager.h]: I'd rather describe it as "Opaque session handle, assigned by PTA".
The only usage I found so far is in `core/arch/arm/tee/pta_socket.c` and that could be replaced by other means.
It seems this field could be made redundant with not too much effort.

Comment (etienne-lms) [reply core/include/kernel/tee_ta_manager.h]: Then maybe better remove this field and update `pta_socket.c` accordingly. Ok?

Comment (jenswi-linaro) [reply core/include/kernel/tee_ta_manager.h]: Or we could do both, adding an accurate comment here is an improvement regardless.

Comment (etienne-lms) [reply core/include/kernel/tee_ta_manager.h]: I'll wait https://github.com/OP-TEE/optee_os/pull/2815/commits/f8f9dc947f5508d6c4140b3930cf4efce981f0f0 is merged before proposing an update.

Comment (etienne-lms) [reply core/include/kernel/tee_ta_manager.h]: Ok, I'll update this P-R according to your comment and postpone `user_ctx` field removal to a later P-R.

**Comment by etienne-lms** [comment]
Pushed a fixup commit for @jenswi-linaro's comment.

Added 2 other commits to fixup some comments trace. Not really related to this P-R description, but there are small changes and should be quickly acked or rejected..

No tag applied yet.

**Comment by jenswi-linaro** [comment]
I've always abbreviated pseudo trusted application as "PTA", with a capital P since it represents a word in the abbreviation.

**Comment by etienne-lms** [comment]
> I've always abbreviated pseudo trusted application as "PTA", with a capital P since it represents a word in the abbreviation.

Ok, I added a fixup commit to change occurrences of pTA into PTA.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Fixup commits squashed and tag applied.

**Pull request merged**